### PR TITLE
CrmAuthenticable to CiviCrm

### PR DESCRIPTION
**Pull request by isaacmg410**

### CODE REVIEW

**Comment by isaacmg410** [comment]
@tramuntanal @aitorlb this PR is an initial state. 
Specs are missing, and some text refactors pending

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.5.3
```
Comment (tramuntanal) [inline .ruby-version]: better latest Ruby version: 2.6.3

File: README.md [hunk 2]
```diff
@@ -49,7 +51,7 @@ crm_login:
 
 ## Versioning
 
-`Decidim::Erc::CrmLogin` depends directly on `Decidim::Core` in `0.18.0` version.
+`Decidim::Erc::CrmAuthenticable` depends directly on `Decidim::Core` in `0.18.0` version.
```
Comment (tramuntanal) [inline README.md]: Gemfile says v0.17

File: app/controllers/decidim/devise/registrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      def new_step_2
+        @form = form(Decidim::Erc::CrmAuthenticable::RegistrationFormStepOneForm).from_params(params[:user])
+        
+        if @form.valid?
+          @form = form(Decidim::Erc::CrmAuthenticable::RegistrationFormStepTwoForm).from_params(
+            user: {
+              sign_up_as: "user",
+            }.merge(extra_params)
+          )
+          flash[:notice] =  "S'ha trobat l'usuari"
```
Comment (tramuntanal) [inline app/controllers/decidim/devise/registrations_controller.rb]: only catalan is ok?

File: app/decorators/decidim/create_registration_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_dependency 'decidim/create_registration'
+Decidim::CreateRegistration.class_eval do
```
Comment (tramuntanal) [inline app/decorators/decidim/create_registration_decorator.rb]: Can you please what is this decoratori overriding?
I mean, explain what it is doing different from the original command.

File: app/decorators/decidim/user_decorator.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_dependency 'decidim/user'
+Decidim::User.class_eval do
```
Comment (tramuntanal) [inline app/decorators/decidim/user_decorator.rb]: Can you please what is this decorator overriding?
I mean, explain what it is doing different from the original class.
If all are helper methods you can explain simply that.

**Review by tramuntanal — COMMENTED** [review]
I'll start with some comments and when we resolve them i'll continue :D

File: app/services/decidim/erc/crm_authenticable/crm_authenticable_action_authorizer.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        protected
+
+        # Method overrided
+        # Handles the verification of the authorization option :join_field.
+        # Converts the required value (membership seniority in number of months)
+        # from Integer to Date and compares it to the metadata field value as Date.
+        def unmatched_fields
+          @unmatched_fields ||= (valued_options_keys & authorization.metadata.to_h.keys).each_with_object({}) do |field, unmatched|
+            if field == "join_date"
+              required_value = (Date.current - options[field].to_i.months)
```
Comment (isaacmg410) [inline app/services/decidim/erc/crm_authenticable/crm_authenticable_action_authorizer.rb]: Ho poso com a nota, aquí crec que haurem de canviar el valor Date.current per la Data del plenari. Ho revisem abans de fer res.

**Review by aitorlb — APPROVED** [review]

**Pull request merged**